Frameworks/CoreText/CTLine.mm

```diff
@@ -22,25 +22,47 @@
 #import <CoreText/CTTypesetter.h>
 #import <CoreFoundation/CFString.h>
 
-#include <memory>
+#import <memory>
 #import <algorithm>
 #import <numeric>
 #import <cwchar>
 #import <vector>
 
+// Adds the characters and attributes of a given attributed string to the end of the receiver.
+static inline void __CFAttributedStringAppendAttributedString(CFMutableAttributedStringRef attrString, CFAttributedStringRef appendString) {
+    CFRange replaceRange = { CFAttributedStringGetLength(attrString), 0 };
+    CFAttributedStringReplaceAttributedString(attrString, replaceRange, appendString);
+}
+
+// Inserts the characters and attribute of the given string into the receiver at the given index.
+static inline void __CFAttributedStringInsertAttributedStringOneAttribute(CFMutableAttributedStringRef attrString,
```

Comment:
I don't think these would provide much perf improvement over your other non-OneAttribute helper, would they? I would like to just get rid of them.